src/train.py

```diff
@@ -0,0 +1,356 @@
+    :param emb_size: embedding size received by visual representations (default: 768)
+    :param num_workers: number of threads in DataLoader (default: 1)
+    :param save_to: the path where the weights of the model are saved after training, if a value is specified, if not, then the weights are not saved (default: None)
+    :param font: font used to generate character images (default: resources/fonts/NotoSans.ttf)
+    :param out_channels: number of output channels in the last convolutional layer (default: 256)
+    :param nhead: number of transformer layer heads (default: 12)
+    :param beta1: Adam optimizer parameter (default: 0.9)
+    :param beta2: Adam optimizer parameter (default: 0.999)
+    :param sl: if true, then the task of labeling the sequence is being solved (default: False)
+    :param dropout: dropout probability (default: 0.0)
+    :param device: 'cuda' or 'cpu'. If None it will be set automatically (default: None)
+    :param warmup: number of steps to warm up Linear Scheduler (default: 1000)
+    :param batch_size: batch size (default: 32)
+    :param kernel_size: convolution kernel size (default 3)
+    :param num_layers: number of transformer encoder layers in the visual representation model (default: 1)
+    :param lr: learning rate (default: 5e-5)
+    :return:
+    """
+    wandb.init(project="visual-text", entity="borisshapa")
+    set_deterministic_mode(random_state)
+
+    actual_device = device or ("cuda" if torch.cuda.is_available() else "cpu")
+    labeled_texts = load_json(train_data)
+
+    val_labeled_texts = None
+    if val_data is not None:
+        val_labeled_texts = load_json(val_data)
+
+    test_labeled_texts = None
+    if test_data is not None:
+        test_labeled_texts = load_json(test_data)
+
+    model: nn.Module
+    criterion: Callable[[torch.Tensor, torch.Tensor], torch.Tensor]
+
+    dataset: SizedCollatedDataset
+    val_dataset: Optional[SizedCollatedDataset] = None
+    test_dataset: Optional[SizedCollatedDataset] = None
+
+    if sl:
+        criterion = BceLossForTokenClassification()
+        create_dataset_sl_f: Callable[
+            [List[Dict[str, Union[List[List[Union[str, int]]], int]]]],
+            SizedCollatedDataset,
+        ]
+        if not tokenizer:
+            model = VisualTextSequenceLabeler(
+                height=font_size,
+                width=window_size,
+                kernel_size=kernel_size,
+                emb_size=emb_size,
+                num_layers=num_layers,
+                out_channels=out_channels,
+                nhead=nhead,
+                dropout=dropout,
+            )
+            create_dataset_sl_f = lambda texts: VTRDatasetSL(
+                labeled_texts=texts,
+                font=font,
+                font_size=font_size,
+                window_size=window_size,
+                stride=stride,
+                max_seq_len=max_seq_len,
+                max_slices_count_per_word=max_slices_count_per_word,
+            )
```

Comment:
I think it is better to use `functools.partial` for such cases rather than anonymous functions.